pallets/liquidity-pools-gateway/src/message_processing.rs

```diff
@@ -14,24 +14,21 @@ use sp_arithmetic::traits::{EnsureAddAssign, EnsureSub};
 use sp_runtime::DispatchError;
 
 use crate::{
-	message::GatewayMessage, Config, Error, InvalidMessageSessions, Pallet, PendingInboundEntries,
-	Routers, SessionIdStore,
+	message::GatewayMessage, Config, Error, Pallet, PendingInboundEntries, Routers, SessionIdStore,
 };
 
-/// The limit used when clearing the `PendingInboundEntries` for invalid
-/// session IDs.
-const INVALID_ID_REMOVAL_LIMIT: u32 = 100;
-
 /// Type used when storing inbound message information.
 #[derive(Debug, Encode, Decode, Clone, Eq, MaxEncodedLen, PartialEq, TypeInfo)]
 #[scale_info(skip_type_params(T))]
 pub enum InboundEntry<T: Config> {
 	Message {
+		session_id: T::SessionId,
 		domain_address: DomainAddress,
 		message: T::Message,
 		expected_proof_count: u32,
```

Comment:
Could not always the `expected_proof_count` be infered by knowing the number of routers under `domain_address`?

Reply:
Yes, but if we want to support duplicate messages and ensure they don't get lost, we need to keep track of this and increment it accordingly, as per [this](https://github.com/centrifuge/centrifuge-chain/blob/feat/lp-v2-gateway-multi-router-drop-session-id-invalidation/pallets/liquidity-pools-gateway/src/message_processing.rs#L178).